### ISSUE TITLE
[Bug]: Storybook 9 - Addon a11y fails to run in monorepos with pnpm

### ISSUE THREAD

**Issue by yannbf** [storybookjs/storybook#31299]
### Describe the bug

When upgrading a project to use Storybook 9 (https://github.com/Endava/BEEQ), the a11y panel shows the following issue:
![Image](https://github.com/user-attachments/assets/315b7a41-356c-4acd-b2d9-11bbd014fd18)

Which is derived from this line of code:
https://github.com/storybookjs/storybook/blob/930f7e52e7e704aebc24f87cb874afc9efd03dac/code/addons/a11y/src/a11yRunner.ts#L44

Seems there must be something going on when resolving axe-core in PNPM projects, which returns undefined for some reason. When installing `axe-core` directly as a dependency of the project, then the issue is fixed.

**Comment by thebuilder** [comment]
Running into the same issue in the 9.0 beta releases - Some details.

- Using **pnpm** (`10.8.0`)
- Not running a mono repo
- Adding `axe-core` as a project dependency doesn't resolve the issue
- Using `@storybook/nextjs-vite`
- Executing the test runner with a11y enabled works correctly.
